src/server/gc_worker/compaction_filter.rs

```diff
@@ -373,7 +373,18 @@ impl WriteCompactionFilter {
 
     // It's possible that elder versions than a MVCC-delete mark occurs in a higher level.
     // So those versions needs to be handled by scan + delete.
-    fn handle_delete_mark(&mut self) -> Result<(), String> {
+    fn handle_delete_mark(&mut self) {
+        if let Err(e) = self.handle_delete_mark_impl() {
+            error!(
+                "WriteCompactionFilter meets a fatal error and can't recover";
+                "err" => ?e,
+            );
+            // Because the section is executed in RocksDB compaction threads,
+            // `exit` seems better than `panic`.
+            unsafe { libc::exit(-1) };
```

Comment:
optional: it seems reasonable to send a PR to rocksdb upstream to add a compaction filter finish callback, with a status that can fail compaction.

Reply:
It's better to handle errors in compaction filter. I will try it later in RocksDB.